packages/dialect-react-ui/components/Chat/index.tsx

```diff
@@ -57,19 +74,31 @@ function InnerChat({
           showThreadSettings(navigate, threadId),
       },
     });
-  }, [configure, disconnectedFromChain, isWalletConnected, navigate]);
+  }, [configure, isSolanaConnected, isWalletConnected, navigate]);
+
+  const someBackendConnected =
+    (isSolanaShouldConnect && isSolanaConnected) ||
+    (isDialectCloudShouldConnect && isDialectCloudConnected);
 
   useEffect(
     function pickRoute() {
-      if (disconnectedFromChain) {
+      if (!someBackendConnected) {
         navigate(RouteName.NoConnection);
       } else if (!isWalletConnected) {
         navigate(RouteName.NoWallet);
+      } else if (isSigning) {
```

Comment:
To clarify, does this navigation happen on ANY transaction signing or only the one that needs for authorization?